api-error-response_test.go

```diff
@@ -182,29 +184,31 @@ func TestHttpRespToErrorResponse(t *testing.T) {
 func TestErrEntityTooLarge(t *testing.T) {
 	msg := fmt.Sprintf("Your proposed upload size ‘%d’ exceeds the maximum allowed object size ‘%d’ for single PUT operation.", 1000000, 99999)
 	expectedResult := ErrorResponse{
+		StatusCode: http.StatusBadRequest,
 		Code:       "EntityTooLarge",
 		Message:    msg,
 		BucketName: "minio-bucket",
 		Key:        "Asia/",
 	}
 	actualResult := ErrEntityTooLarge(1000000, 99999, "minio-bucket", "Asia/")
 	if !reflect.DeepEqual(expectedResult, actualResult) {
-		t.Errorf("Expected result to be '%+v', but instead got '%+v'", expectedResult, actualResult)
+		t.Errorf("Expected result to be '%#v', but instead got '%#v'", expectedResult, actualResult)
```

Comment:
`%#v` instead of `%+v` add lot of information and feels verbose IMO. 

Here is a sample run with `%+v` 

```
$: Expected result to be 'Your proposed upload size ‘1000000’ exceeds the maximum allowed object size ‘99999’ for single PUT operation.', but instead got 'Your proposed upload size ‘1000000’ exceeds the maximum allowed object size ‘199999’ for single PUT operation.'
```

While with `%#v`

```
$: Expected result to be 'minio.ErrorResponse{XMLName:xml.Name{Space:"", Local:""}, StatusCode:400, Code:"EntityTooLarge", Message:"Your proposed upload size ‘1000000’ exceeds the maximum allowed object size ‘99999’ for single PUT operation.", BucketName:"minio-bucket", Key:"Asia/", RequestID:"", HostID:"", Region:"", Headers:http.Header(nil)}', but instead got 'minio.ErrorResponse{XMLName:xml.Name{Space:"", Local:""}, StatusCode:400, Code:"EntityTooLarge", Message:"Your proposed upload size ‘1000000’ exceeds the maximum allowed object size ‘199999’ for single PUT operation.", BucketName:"minio-bucket", Key:"Asia/", RequestID:"", HostID:"", Region:"", Headers:http.Header(nil)}'
```
Is there any specific benefit of doing this?

Reply:
@nitisht since the reflect.DeepEqual might not match for lot of fields other than the message itself, having #v is okay in this scenario. Since as a mismatch should be more noticeable.

Reply:
:+1:

Reply:
Yeah, the reason I added it was that it made it a little harder to debug an error, since %+v only returns the String() form, which doesn't include the status code or other fields that might be mismatched. In my own projects I use `gomega` which does this automatically.